### PR TITLE
Login functionality

### LINKED ISSUE #2: As a user, I want to log in securely so that I can access my account and interact with the blogs. (BE)
1. Code Quality
- [x] Is the code easy to read and understand?
- [x] Are functions/methods short, clear, and focused on a single task?
- [x] Is the code modular and reusable where appropriate?

2. Functionality
- [x] Does the code meet all the functional requirements?
- [x] Are all edge cases and error conditions handled?
- [x] Does it integrate correctly with existing code?

3. Style & Conventions
- [x] Does the code follow the team’s style guide (e.g., PEP8 for Python, Airbnb for JavaScript)?
- [x] Are naming conventions consistent and meaningful?
- [x] Are indentation and spacing consistent?

4. Testing
- [x] Are there unit and/or integration tests for new features?
- [x] Do the tests cover both normal and edge cases?
- [x] Do all tests pass?

5. Security
- [x] Are all user inputs properly validated and sanitized?
- [x] Are there any hardcoded credentials or secrets?
- [x] Is sensitive data handled securely?

6. Performance
- [x] Is the code efficient and optimized where needed?
- [x] Are there any unnecessary computations or database queries?
- [x] Are large datasets handled properly?

7. Documentation
- [x] Are code comments used where necessary to explain complex logic?
- [x] Are public methods and APIs documented?
- [x] Is the README or other project documentation updated (if needed)?

### PR DESCRIPTION
**Pull request by goonle**
1. Code Quality

- [x] Is the code easy to read and understand?

- [x] Are functions/methods short, clear, and focused on a single task?

- [x] Is the code modular and reusable where appropriate?

2. Functionality

- [x] Does the code meet all the functional requirements?

- [x] Are all edge cases and error conditions handled?

- [x] Does it integrate correctly with existing code?

3. Style & Conventions

- [x] Does the code follow the team’s style guide (e.g., PEP8 for Python, Airbnb for JavaScript)?

- [x] Are naming conventions consistent and meaningful?

- [x] Are indentation and spacing consistent?

4. Testing

- [x] Are there unit and/or integration tests for new features?

- [x] Do the tests cover both normal and edge cases?

- [x] Do all tests pass?

5. Security

- [x] Are all user inputs properly validated and sanitized?

- [x] Are there any hardcoded credentials or secrets?

- [x] Is sensitive data handled securely?

6. Performance

- [x] Is the code efficient and optimized where needed?

- [x] Are there any unnecessary computations or database queries?

- [x] Are large datasets handled properly?

7. Documentation

- [x] Are code comments used where necessary to explain complex logic?

- [x] Are public methods and APIs documented?

- [ ] Is the README or other project documentation updated (if needed)?

### CODE REVIEW

File: blogs_api/views.py [hunk 1]
```diff
@@ -27,3 +28,24 @@ def post(self, request):
...
+class LoginAPIView(APIView):
+    permission_classes = [AllowAny]
+
+    @swagger_auto_schema(
+        request_body=LoginSerializer,
+        responses={200: "Login successful", 400: "Invalid credentials"},
+    )
+    def post(self, request):
+        serializer = LoginSerializer(data=request.data)
+        if serializer.is_valid():
```
Comment (mfasl2k) [inline blogs_api/views.py:42]: return a Response when is not valid

Comment (goonle) [reply blogs_api/views.py:42]: Yes fixed

**Review by mfasl2k — APPROVED** [review]
